### PR TITLE
DYN-6162 Adding Lucene Search Tests

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Adding more unit tests for Lucene Search
As a part of this change I've re-enabled the  WhenStartingDynamoOperatorNodesNolongerMissingFromSearch() test and also I've added other test like for checking all the nodes, categories and search packages in other languages.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

Adding more unit tests for Lucene Search

### Reviewers

@QilongTang

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
Screenshot showing that the tests added are passing locally.
image
image

File: test/DynamoCoreWpfTests/SearchSideEffects.cs [hunk 1]
```diff
@@ -87,5 +80,63 @@ public void WhenCustomNodeWorkspaceIsFocusedInputAndOutputNodesArePresentInSearc
...
+            string category = "Core.Input";
+
+            // Search and check that the results are correct based in the node name provided for the searchTerm
+            var nodesResult = ViewModel.CurrentSpaceViewModel.InCanvasSearchViewModel.Search(category,true);
+
+            Assert.That(nodesResult.Count(), Is.GreaterThan(0));
+
+            foreach (var node in nodesResult.Take(10))
+            {
+                Assert.IsTrue(node.Category.Contains(category.Split('.')[0]) || node.Category.Contains(category.Split('.')[1]));
```
Comment (QilongTang) [inline test/DynamoCoreWpfTests/SearchSideEffects.cs:138]: Comments for this line please

**Review by QilongTang — APPROVED** [review]

File: test/DynamoCoreWpfTests/SearchSideEffects.cs (hunk 1 above)
Comment (RobertGlobant20) [reply test/DynamoCoreWpfTests/SearchSideEffects.cs:141]: I've added more comments in the next commit: https://github.com/DynamoDS/Dynamo/pull/14401/commits/a4f79e14156fbcde266db4eddee69b591bd345a8
Please review it. Thanks

**Comment by QilongTang** [comment]
Thank you @RobertGlobant20 These tests are good, however, I do not see a test covering search results order. Can you make at least one covering, let's say search for a term and then validate the order? name -> category -> description etc.

**Comment by RobertGlobant20** [comment]
> Thank you @RobertGlobant20 These tests are good, however, I do not see a test covering search results order. Can you make at least one covering, let's say search for a term and then validate the order? name -> category -> description etc.
> 
> Or if you prefer, we can use another PR and get this one in.

I've added a new test that just validates the order of two searchTerms please let me know if it looks to you or if something else needs to be added.

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
Regression [Dynamo.PackageManager.Wpf.Tests.PackageManagerSearchElementViewModelTests.PackageSearchDialogSearchDifferentLanguage](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/11151/testReport/junit/Dynamo.PackageManager.Wpf.Tests/PackageManagerSearchElementViewModelTests/PackageSearchDialogSearchDifferentLanguage/)

**Comment by QilongTang** [comment]
@RobertGlobant20 Please also take a look at the merge conflicts

**Pull request merged**